addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -1514,6 +1515,23 @@ def test_message_process_file_encoding_ascii(self):
         # This explains the multiple "�" in the attachment.
         self.assertIn("Chauss������e de Bruxelles", record.message_main_attachment_id.raw.decode())
 
+    def test_message_route_reply_model_none(self):
+        message_id = f'<test{time.time()}-openerp@{socket.gethostname()}>'
+        self.env['mail.message'].create({
+            'email_from': self.email_from,
+            'body': '<p>test</p>',
+            'message_id': message_id,
+            'message_type': 'email',
+            'model': None,
+            'res_id': None,
+        })
+        self.env['mail.alias'].create({'alias_name': 'test', 'alias_model_id': self.env['ir.model']._get('res.partner').id})
+        record = self.format_and_process(
+            MAIL_TEMPLATE, self.email_from, 'groups@test.com',
+            subject='Spammy', extra='In-Reply-To:\r\n\t%s\n' % message_id, model=None)
+        self.assertEqual(len(record) > 0, True)
```

Comment:
```suggestion
        self.assertGreater(len(record), 0)
```
or
```
      self.assertTrue(record)
```

I would also check the record is of the right type, just to be safe.

Reply:
if it returns a record it will be of type `mail.test.gateway` right? because of this line in `format_and_process` 
```
return self.env[target_model].search([(target_field, '=', subject)])
```
@thiryr